packages/amplify_api_plugin_interface/pubspec.yaml

```diff
@@ -14,6 +14,9 @@ dependencies:
   meta: ^1.3.0
   uuid: ^3.0.1
 
+  # TODO: Datastore dependencies temporarily added in API. Eventually they should be moved to core or otherwise reconciled to avoid duplication.
```

Comment:
I would think this is the only place this needs to be added as it's a dependency for API plugin, which is a dep for the example app. Could you try removing the other additions to pubspec to see if it works as expected?

Reply:
I tried removing it as a dependency from the locations you specified, but imports are unable to resolve to amplify_api_plugin_interface. My hunch is that's due to only importing the datastore interface & not exporting the types too. That being said I think this is better because it doesn't allow the types to propagate down and cause potential issues like I saw with previous approaches. Additionally Datastore is an optional dependency.

Reply:
Cool, thanks for looking into this. It seems weird, but I guess as a temporary solution it's fine and we can sort this out when we eventually resolve this in a shared place.